src/js/Color.js

```diff
@@ -0,0 +1,537 @@
+        var a = 1;
+        return new axs.color.Color(r, g, b, a);
+    }
+
+    var rgbaRegex = /^rgba\((\d+), (\d+), (\d+), (\d*(\.\d+)?)\)/;
+    match = colorString.match(rgbaRegex);
+    if (match) {
+        var r = parseInt(match[1], 10);
+        var g = parseInt(match[2], 10);
+        var b = parseInt(match[3], 10);
+        var a = parseFloat(match[4]);
+        return new axs.color.Color(r, g, b, a);
+    }
+
+    return null;
+};
+
+/**
+ * @param {number} value The value of a color channel, 0 <= value <= 0xFF
+ * @return {!string}
+ */
+axs.color.colorChannelToString = function(value) {
+    value = Math.round(value);
+    if (value <= 0xF)
+        return '0' + value.toString(16);
+    return value.toString(16);
+};
+
+/**
+ * @param {axs.color.Color} color
+ * @return {!string}
+ */
+axs.color.colorToString = function(color) {
+    if (color.alpha == 1) {
+         return '#' + axs.color.colorChannelToString(color.red) +
+         axs.color.colorChannelToString(color.green) + axs.color.colorChannelToString(color.blue);
+    }
+    else
+        return 'rgba(' + [color.red, color.green, color.blue, color.alpha].join(',') + ')';
+};
+
+/**
+ * Compute a desired luminance given a given luminance and a desired contrast ratio.
+ * @param {number} luminance The given luminance.
+ * @param {number} contrast The desired contrast ratio.
+ * @param {boolean} higher Whether the desired luminance is higher or lower than the given luminance.
+ * @return {number} The desired luminance.
+ */
+axs.color.luminanceFromContrastRatio = function(luminance, contrast, higher) {
+    if (higher) {
+        var newLuminance = (luminance + 0.05) * contrast - 0.05;
+        return newLuminance;
+    } else {
+        var newLuminance = (luminance + 0.05) / contrast - 0.05;
+        return newLuminance;
+    }
+};
+
+/**
+ * Given a color in YCC format and a desired luminance, pick a new color with the desired luminance which is
+ * as close as possible to the original color.
+ * @param {axs.color.YCC} ycc The original color in YCC form.
+ * @param {number} luma The desired luminance
+ * @return {!axs.color.Color} A new color in RGB.
+ */
+axs.color.translateColor = function(ycc, luma) {
+    var endpoint = (luma > ycc.luma) ? axs.color.WHITE_YCC : axs.color.BLACK_YCC;
+    var cubeFaces = (endpoint == axs.color.WHITE_YCC) ? axs.color.YCC_CUBE_FACES_WHITE
+                                                      : axs.color.YCC_CUBE_FACES_BLACK;
+
+    var a = new axs.color.YCC([0, ycc.Cb, ycc.Cr]);
+    var b = new axs.color.YCC([1, ycc.Cb, ycc.Cr]);
+    var line = { a: a, b: b };
+
+    var intersection = null;
+    for (var i = 0; i < cubeFaces.length; i++) {
+        var cubeFace = cubeFaces[i];
+        intersection = axs.color.findIntersection(line, cubeFace);
+        // If intersection within [0, 1] in Z axis, it is within the cube.
+        if (intersection.z >= 0 && intersection.z <= 1)
+            break;
+    }
+    if (!intersection) {
```

Comment:
To be honest I don't know. I'm pretty sure it shouldn't, but I'd rather it fail non-mysteriously than mysteriously if it does. Made it an exception, since it's an exceptional case.